bittensor/_neuron/text/core_server/nucleus_impl.py

```diff
@@ -362,42 +356,37 @@ def encode_forward_causallm(self, token_batch, tokenizer=None, encode_len=bitten
-                                                    output_hidden_states=True)
+        def _forward(_model_output=model_output):
+            if _model_output is None:
+                # transformer models like gerpt2 typically perform worse with left-side attention mask, so turning it off
+                _model_output = self.pre_model(input_ids=tokens['input_ids'],
+                                               # attention_mask=tokens['attention_mask'],
+                                               output_hidden_states=True)
 
-        pre_logits = model_output.logits
+            pre_logits = _model_output.logits  # [batch_size, sequence_len, self.tokenizer.vocab_len]
 
-        if self.config.neuron.remote_train:
             probs_std = translate_logits_to_probs_std(pre_logits,
-                                                        tokens['offset_mapping'], tokens['offset_mapping_std'],
-                                                        self.tokenizer, self.std_tokenizer,
-                                                        self.split_map_cache,
-                                                        self.to_translation_map, self.from_translation_map,
-                                                        tokens['input_ids'], token_batch)
-        
-        else:
-            with torch.no_grad():
-                probs_std = translate_logits_to_probs_std(pre_logits,
-                                                            tokens['offset_mapping'], tokens['offset_mapping_std'],
-                                                            self.tokenizer, self.std_tokenizer,
-                                                            self.split_map_cache,
-                                                            self.to_translation_map, self.from_translation_map,
-                                                            tokens['input_ids'], token_batch)
-        probs_std = probs_std.to(self.device)
-        logits_std = torch.log(probs_std + 1e-40)
-
-        return model_output, logits_std
+                                                      tokens['offset_mapping'], tokens['offset_mapping_std'],
+                                                      self.tokenizer, self.std_tokenizer,
+                                                      self.split_map_cache,
+                                                      self.to_translation_map, self.from_translation_map,
+                                                      tokens['input_ids'], token_batch)
+            probs_std = probs_std.to(self.device)
+            logits_std = torch.log(probs_std + 1e-40)
+
+            original_loss = self.get_loss_fct(pre_logits, tokens['input_ids'])
+            translated_loss = self.get_loss_fct(logits_std, token_batch)
+            logger.info(f'TextCausalLM \t| Server loss: {original_loss: .2f} \t| Translated loss: {translated_loss: .2f}')
```

Comment:
Out of curiosity, how much is the difference in loss?

Reply:
Translation loss can vary on average between 0.1 - 0.3 nats above original loss.